app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -168,7 +167,18 @@ void addObservations(List<Obs> observations) {
             }
         }
 
-        /** Returns a column for a given date. Creates a new column, if it does not exist yet */
+        /** Ensures that columns are shown for any days in which an order is prescribed. */
+        void addOrders(List<Order> orders) {
+            for (Order order : orders) {
+                if (order.stop != null) {
+                    for (DateTime dt = order.start; !dt.isAfter(order.stop.plusDays(1)); dt = dt.plusDays(1)) {
```

Comment:
Yes, that sounds right!  (I think the UI for this is not that great and we
might want to get rid of the STOP to make it more like the paper charts
they use.  But that's for later.)

On Wed, Nov 4, 2015 at 10:05 AM, Leonardo Lima de Vasconcellos <
notifications@github.com> wrote:

> In app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java
> <https://github.com/projectbuendia/client/pull/72#discussion_r43915620>:
>
> > @@ -168,7 +167,18 @@ void addObservations(List<Obs> observations) {
> >              }
> >          }
> >
> > -        /** Returns a column for a given date. Creates a new column, if it does not exist yet */
> > +        /** Ensures that columns are shown for any days in which an order is prescribed. */
> > +        void addOrders(List<Order> orders) {
> > +            for (Order order : orders) {
> > +                if (order.stop != null) {
> > +                    for (DateTime dt = order.start; !dt.isAfter(order.stop.plusDays(1)); dt = dt.plusDays(1)) {
>
> @zestyping <https://github.com/zestyping> I added the
> order.stop.plusDays(1) to make the STOP day appear for a future order. Is
> that ok? If so it will be ready to merge!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectbuendia/client/pull/72/files#r43915620>.
>